spark/src/main/scala/org/apache/spark/sql/delta/commands/VacuumCommand.scala

```diff
@@ -125,6 +124,39 @@ object VacuumCommand extends VacuumCommandImpl with Serializable {
     }.toDF("path")
   }
 
+  def getFilesFromReservoir(basePath: String,
+                            partitionColumns: Seq[String],
+                            reservoir: DataFrame): Dataset[SerializableFileStatus] = {
+    val expectedSchema = StructType(
+      Seq(
+        StructField("path", StringType),
+        StructField("length", LongType),
+        StructField("isDir", BooleanType),
+        StructField("modificationTime", LongType)
```

Comment:
Can we put in comments to clarify the meaning? In particular:
- Is the path expected to be a URI, even for local files?
- What is the unit and epoch of modificationTime? (Unix millis or unix micros?)